joomla/joomla-cms
# [4.3] Media manager, refactor the session fallback

Pull Request for Issue # .

### Summary of Changes
- Refactor the fallback code for the url/session initial path handling

### Testing Instructions
- test a url like (existing path) `/administrator/index.php?option=com_media&path=local-images:/banners`
- test a url like (not existing path) `/administrator/index.php?option=com_media&path=local-none:/banners`
- test a url like (not existing path) `/administrator/index.php?option=com_media&path=local-none:/noooooo`
- Open the browser console and edit the session value from something like: `{"selectedDirectory":"local-images:/banners","showInfoBar":false,"listView":"grid","gridSize":"md","search":"","sortBy":"name","sortDirection":"asc"}` to values matching the changes on the previous steps (eg `"selectedDirectory":"local-none:/banners"`, "selectedDirectory":"local-none:/noooooo") and check if the media manager loads without an error
- Apply all possible combos between the url and the session storage

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Link to documentations
Please select:



@laoneo could you review this one?
